### PR TITLE
Handle write on identifyTime attribute on esp32 side

### PR DESCRIPTION
**Pull request by PSONALl**
#### What is being fixed?  
Handled attribute write command for identifyTime attribute on esp32 side in all-clusters-app 

#### Change overview
Added OnIdentifyAttributeChangeCallback in DeviceCallbacks.cpp

#### Testing
* Tested commissioning and cluster control using chip-tool

### CODE REVIEW

**Comment by woody-apple** [comment]
Fast tracking, given platform changes by a platform maintainer.

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@PSONALl please restyle this PR! :)

**Review by dhrishi — APPROVED** [review]

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 1]
```diff
@@ -254,6 +258,15 @@ void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointI
 }
 #endif
 
+void DeviceCallbacks::OnIdentifyAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+        if (attributeId == ZCL_IDENTIFY_TIME_ATTRIBUTE_ID)
+        {
+		uint8_t identifyTime = *value;
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:265]: Please fix formatting or restyle

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 2]
```diff
@@ -156,6 +157,9 @@ void DeviceCallbacks::PostAttributeChangeCallback(EndpointId endpointId, Cluster
         OnColorControlAttributeChangeCallback(endpointId, attributeId, value);
         break;
 #endif
+    case ZCL_IDENTIFY_CLUSTER_ID:
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:160]: Please use the namespaced cluster ids.

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 3]
```diff
@@ -254,6 +258,15 @@ void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointI
 }
 #endif
 
+void DeviceCallbacks::OnIdentifyAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+        if (attributeId == ZCL_IDENTIFY_TIME_ATTRIBUTE_ID)
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:263]: Please use the namespaced attribute ids.

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 4]
```diff
@@ -254,6 +258,15 @@ void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointI
...
+        {
+		uint8_t identifyTime = *value;
+            emberAfWriteServerAttribute(endpointId, ZCL_IDENTIFY_CLUSTER_ID,ZCL_IDENTIFY_TIME_ATTRIBUTE_ID, &identifyTime, ZCL_INT8U_ATTRIBUTE_TYPE);
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:266]: I don't understand this line.  We are being notified that the attribute changed.  Our response is to ... write the attribute?  Won't that just end up calling this same code recursively?

What is this PR actually trying to do?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp (hunk 1 above)
Comment (PSONALl) [reply examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp]: restyled

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 5]
```diff
@@ -254,6 +258,26 @@ void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointI
 }
 #endif
 
+void DeviceCallbacks::OnIdentifyPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+    if (attributeId == ZCL_IDENTIFY_TIME_ATTRIBUTE_ID)
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:263]: Please use namespaced attribute ids.

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp (hunk 2 above)
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:160]: Pleaes use namespaced cluster ids.

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 6]
```diff
@@ -254,6 +258,26 @@ void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointI
...
+    {
+        uint8_t identifyTime = *value;
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:265]: This does not look right to me.  IdentifyTime is of type `int16u`, no?

`PostAttributeChangeCallback` was passed the size of the data to read from `value`.  We're ignoring that size, and reading the wrong amount of data....

Instead we should be passing the size in here, checking that it matches expectations (i.e. is 2), and if so reading the value as a `uint16_t`, no?

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 7]
```diff
@@ -254,6 +258,26 @@ void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointI
...
+        bool onOffState;
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:266]: Please declare this boolean in the block where it's used?

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp (hunk 2 above)
Comment (PSONALl) [reply examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp]: Added

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp (hunk 6 above)
Comment (PSONALl) [reply examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp]: added this change

**Comment by woody-apple** [comment]
Fast tracking, given > 3 days for review.

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 8]
```diff
@@ -139,23 +141,32 @@ void DeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, intptr_
...
-    case ZCL_COLOR_CONTROL_CLUSTER_ID:
+    case Clusters::ColorControl::Id:
         OnColorControlAttributeChangeCallback(endpointId, attributeId, value);
         break;
 #endif
+    case Clusters::Identify::Id:
+        /*
+        if( size == 2 )
+          else
+        */
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:167]: Why this comment?

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 9]
```diff
@@ -254,6 +266,28 @@ void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointI
 }
 #endif
 
+void DeviceCallbacks::OnIdentifyPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint16_t size,
+                                                            uint8_t * value)
+{
+    if (attributeId == Clusters::Identify::Attributes::IdentifyTime::Id)
+    {
+        uint16_t identifyTime;
+        size == 2 ? identifyTime = (value[1] << 8) + value[0] : identifyTime = value[0];
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:275]: No, if `size != 2` then this is some sort of broken call, and you can't assume anything about the value....

Furthermore, if `size == 2`, then `value` points to a native-endian `uint16_t`, whereas this code is assuming little-endian.

In other words, once we check size == 2, the next step is:
```
    uint16_t identifyTime;
    memcpy(&identifyTime, value, size);
```

File: examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp [hunk 10]
```diff
@@ -254,6 +266,28 @@ void DeviceCallbacks::OnColorControlAttributeChangeCallback(EndpointId endpointI
...
+        if (identifyTime)
+        {
+            // Currently we have no separate indicator LEDs on each endpoints.
+            // We are using LED1 for endpoint 0,1 and LED2 for endpoint 2
+            endpointId == 2 ? statusLED2.Blink(kIdentifyTimerDelayMS * 2) : statusLED1.Blink(kIdentifyTimerDelayMS * 2);
+        }
+        else
+        {
+            bool onOffState;
+            endpointId == 0 ? onOffState = mEndpointOnOffState[0] : onOffState = mEndpointOnOffState[endpointId - 1];
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/DeviceCallbacks.cpp:285]: Might be more readable as:
 ```suggestion
            bool onOffState = mEndpointOnOffState[endpointId == 0 ? 0 : endpointId - 1];
```
but don't feel too strongly about it.

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**